backend/src/main/java/de/bund/digitalservice/ris/norms/adapter/input/restapi/formatter/XmlFormattingResponseBodyAdvice.java

```diff
@@ -0,0 +1,80 @@
+package de.bund.digitalservice.ris.norms.adapter.input.restapi.formatter;
+
+import java.io.StringReader;
+import java.io.StringWriter;
+import javax.xml.XMLConstants;
+import javax.xml.parsers.DocumentBuilderFactory;
+import org.springframework.core.MethodParameter;
+import org.springframework.http.MediaType;
+import org.springframework.http.converter.HttpMessageConverter;
+import org.springframework.http.server.ServerHttpRequest;
+import org.springframework.http.server.ServerHttpResponse;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseBodyAdvice;
+import org.w3c.dom.Document;
+import org.w3c.dom.ls.DOMImplementationLS;
+import org.w3c.dom.ls.LSOutput;
+import org.w3c.dom.ls.LSSerializer;
+import org.xml.sax.InputSource;
+
+/**
+ * ResponseBodyAdvice that formats XML responses
+ */
+@ControllerAdvice
+public class XmlFormattingResponseBodyAdvice implements ResponseBodyAdvice<Object> {
+
+  @Override
+  public boolean supports(
+    MethodParameter returnType,
+    Class<? extends HttpMessageConverter<?>> converterType
+  ) {
+    return true;
+  }
+
+  @Override
+  public Object beforeBodyWrite(
+    Object body,
+    MethodParameter returnType,
+    MediaType selectedContentType,
+    Class<? extends HttpMessageConverter<?>> selectedConverterType,
+    ServerHttpRequest request,
+    ServerHttpResponse response
+  ) {
+    if (
+      selectedContentType != null &&
+      selectedContentType.isCompatibleWith(MediaType.APPLICATION_XML) &&
+      body instanceof String xmlBody
+    ) {
+      return formatXml(xmlBody);
+    }
+    return body;
+  }
+
+  private String formatXml(String xml) {
+    try {
+      DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
```

Comment:
To get the `Document` out of the string you could just re-use our `XmlMapper` class, method `toDocument(string)`